### ISSUE TITLE
[BUG]UnicodeDecodeError for flows containing multibyte characters

### ISSUE THREAD

**Issue by d-ataka** [microsoft/promptflow#668]
**Describe the bug**
A problem has been identified concerning flows that include multibyte characters in either the input or jinja.

**How To Reproduce the bug**
Steps to reproduce the bug:
1. Construct a flow that accepts inputs in jinja and outputs them as is, without any processing.
2. The flow will function correctly if the input comprises single-byte characters, but it will fail when dealing with multibyte characters.

**Screenshots**

single-byte characters
![image](https://github.com/microsoft/promptflow/assets/137765645/fd278385-b547-4b8c-aa86-3f3ccfd34f01)

multibyte characters
![image](https://github.com/microsoft/promptflow/assets/137765645/2422ec3f-02c2-420e-b197-e36f11c67e4e)

**Running Information**
 - Promptflow Package Version: 0.1.0b7.post1
 - Operating System: Windows 10
 - Python Version: python==3.9.18

**Comment by d-ataka** [comment]
I used the same version of promptflow and changed the execution environment to WSL2+Ubuntu and it worked. it seems to be a Windows specific problem.
![image](https://github.com/microsoft/promptflow/assets/137765645/e4767c64-0178-45cc-b3f5-d5f1cb76f0ef)

**Comment by ttthree** [comment]
Somehow i was unable to repro issue even with System Locale set to Japanese or Chinese.
@wangchao1230  do you think we should force the files we generate (like flow.dag.yaml) to be UTF-8?

**Comment by d-ataka** [comment]
@ttthree 
I understand this issue pertains to character encoding. Yet, the same problem occurs even when using a file from [basic-with-builtin-llm](https://github.com/microsoft/promptflow/tree/main/examples/flows/standard/basic-with-builtin-llm). 
Intriguingly, it doesn't seem to fail with all multibyte characters.

Inputs that work:

- Python 日本語でこんにちは!
- Python 日本語でHello World!

Inputs that fail:

- Python print 日本語

It's possible that the issue arises specifically when the input ends with the word `日本語`.

```
Traceback (most recent call last):
  File "C:\Users\username\Anaconda3\envs\pf\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\username\Anaconda3\envs\pf\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\username\Anaconda3\envs\pf\lib\site-packages\promptflow\_cli\_pf\entry.py", line 97, in <module>
    main()
  File "C:\Users\username\Anaconda3\envs\pf\lib\site-packages\promptflow\_cli\_pf\entry.py", line 93, in main
    entry(command_args)
  File "C:\Users\username\Anaconda3\envs\pf\lib\site-packages\promptflow\_cli\_pf\entry.py", line 75, in entry
    raise ex
  File "C:\Users\username\Anaconda3\envs\pf\lib\site-packages\promptflow\_cli\_pf\entry.py", line 59, in entry
    dispatch_flow_commands(args)
  File "C:\Users\username\Anaconda3\envs\pf\lib\site-packages\promptflow\_cli\_pf\_flow.py", line 60, in dispatch_flow_commands
    test_flow(args)
  File "C:\Users\username\Anaconda3\envs\pf\lib\site-packages\promptflow\_cli\_utils.py", line 363, in wrapper
    raise e
  File "C:\Users\username\Anaconda3\envs\pf\lib\site-packages\promptflow\_cli\_utils.py", line 351, in wrapper
    return func(*args, **kwargs)
  File "C:\Users\username\Anaconda3\envs\pf\lib\site-packages\promptflow\_cli\_pf\_flow.py", line 344, in test_flow
    result = pf_client.flows._test(
  File "C:\Users\username\Anaconda3\envs\pf\lib\site-packages\promptflow\_sdk\operations\_flow_operations.py", line 99, in _test
    with TestSubmitter(flow=flow, variant=variant).init() as submitter:
  File "C:\Users\username\Anaconda3\envs\pf\lib\contextlib.py", line 119, in __enter__
    return next(self.gen)
  File "C:\Users\username\Anaconda3\envs\pf\lib\site-packages\promptflow\_sdk\operations\_test_submitter.py", line 48, in init
    with variant_overwrite_context(self._origin_flow.code, tuning_node, node_variant) as temp_flow:
  File "C:\Users\username\Anaconda3\envs\pf\lib\contextlib.py", line 119, in __enter__
    return next(self.gen)
  File "C:\Users\username\Anaconda3\envs\pf\lib\site-packages\promptflow\_sdk\operations\_run_submitter.py", line 172, in variant_overwrite_context
    flow_dag_path, _ = _load_flow_dag(flow_path)
  File "C:\Users\username\Anaconda3\envs\pf\lib\site-packages\promptflow\_sdk\operations\_run_submitter.py", line 62, in _load_flow_dag        
    flow_dag = yaml.safe_load(f)
  File "C:\Users\username\Anaconda3\envs\pf\lib\site-packages\yaml\__init__.py", line 125, in safe_load
    return load(stream, SafeLoader)
  File "C:\Users\username\Anaconda3\envs\pf\lib\site-packages\yaml\__init__.py", line 79, in load
    loader = Loader(stream)
  File "C:\Users\username\Anaconda3\envs\pf\lib\site-packages\yaml\loader.py", line 34, in __init__
    Reader.__init__(self, stream)
  File "C:\Users\username\Anaconda3\envs\pf\lib\site-packages\yaml\reader.py", line 85, in __init__
    self.determine_encoding()
  File "C:\Users\username\Anaconda3\envs\pf\lib\site-packages\yaml\reader.py", line 124, in determine_encoding
    self.update_raw()
  File "C:\Users\username\Anaconda3\envs\pf\lib\site-packages\yaml\reader.py", line 178, in update_raw
    data = self.stream.read(size)
UnicodeDecodeError: 'cp932' codec can't decode byte 0x9e in position 205: illegal multibyte sequence
```

**Comment by d-ataka** [comment]
I hope the following information will be helpful.

- In the initial example provided, flow.dag.yaml and jinja2 were saved in utf-8.
- In the initial example provided, a UnicodeDecodeError did not occur when the input was multibyte characters other than `日本語`.
  - For example, if the input is `日本`, no error will occur. However, as shown in the screenshot below, the result will be strange.
  - ![image](https://github.com/microsoft/promptflow/assets/137765645/6fc8bb1b-6819-47df-8e09-71fd4a7ed977)
    - Although the input is `日本`, the input in Jinja is the garbled `譌･譛ｬ`.

**Comment by wangchao1230** [comment]
@ttthree  Yes, I think we should force the encoding of the flow.dag.yaml to be UTF-8. 
So we need to change the places we read/write this file. For sdk we can start by changing places using: yaml.safe_load.

I tried to use some file encoding other thatn utf-8, it will raise strange error for me. 
But I coudn't repro issue when using utf-8 yaml encoding and with 日本語 as input
Adding @0mza987 to follow up for he has more knowledge on 日本語 :)

**Comment by 0mza987** [comment]
I could not repro the codec error but I  re-proed this:
`Although the input is 日本, the input in Jinja is the garbled 譌･譛ｬ.`

It is because for `pf flow test` command there are some places that we should have added the encoding parameters when reading/writing yaml files. I have opened a PR to fix it and I think this should also solve the `UnicodeDecodeError`

https://github.com/microsoft/promptflow/pull/683/files

**Comment by wangchao1230** [comment]
Please note this issue should be resolved in latest version 1.0.0. 
Please let us know if you still met such issue.
